app/controllers/changes_controller.rb

```diff
@@ -5,7 +5,9 @@ def undo
     version = PaperTrail::Version.find(params.require :id)
     # ... and return is correct here
     # rubocop:disable Style/AndOr
-    head :unauthorized and return unless version.done_by? @current_user
+    head :unauthorized and return unless version
+                                         .whodunnit
+                                         .to_i == @current_user.id
```

Comment:
Kind of awkward to read. I might instead decide if the user IDs match in a separate expression.